ddtrace/contrib/starlette/patch.py

```diff
@@ -10,16 +12,58 @@
 log = get_logger(__name__)
 
 config._add(
-    "starlette", dict(_default_service="starlette", request_span_name="starlette.request", distributed_tracing=True)
+    "starlette",
+    dict(
+        _default_service="starlette",
+        request_span_name="starlette.request",
+        distributed_tracing=True,
+        aggregate_resources=True,
+    ),
 )
 
 
+def traced_init(wrapped, instance, args, kwargs):
+    global routes
+    routes = kwargs["routes"]
+    mw = kwargs.pop("middleware", [])
+    mw.insert(0, Middleware(TraceMiddleware, integration_config=config.starlette))
+    kwargs.update({"middleware": mw})
+
+    wrapped(*args, **kwargs)
+
+
+def get_resource(scope):
+    path = None
+    global routes
```

Comment:
We should avoid the `global` variable here and access the routes from `scope["app"].routes`.